extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js

```diff
@@ -0,0 +1,369 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {createIframePromise} from '../../../../testing/iframe';
+import {Services} from '../../../../src/services';
+import {
+  AmpAdNetworkDoubleclickImpl,
+  resetRtcStateForTesting,
+} from '../amp-ad-network-doubleclick-impl';
+import {createElementWithAttributes} from '../../../../src/dom';
+import {installDocService} from '../../../../src/service/ampdoc-impl';
+import {Xhr} from '../../../../src/service/xhr-impl';
+
+// Uncomment out the next line when testing locally
+// import {AmpAd} from '../../../amp-ad/0.1/amp-ad';
+
+function setupForAdTesting(fixture) {
+  installDocService(fixture.win, /* isSingleDoc */ true);
+  const doc = fixture.doc;
+  doc.win = fixture.win;
+  // TODO(a4a-cam@): This is necessary in the short term, until A4A is
+  // smarter about host document styling.  The issue is that it needs to
+  // inherit the AMP runtime style element in order for shadow DOM-enclosed
+  // elements to behave properly.  So we have to set up a minimal one here.
+  const ampStyle = doc.createElement('style');
+  ampStyle.setAttribute('amp-runtime', 'scratch-fortesting');
+  doc.head.appendChild(ampStyle);
+}
+
+
+describes.realWin('DoubleClick Fast Fetch RTC', {amp: true}, env => {
+  let impl;
+  let element;
+  let sandbox;
+  let xhrMock;
+
+  beforeEach(() => {
+    sandbox = env.sandbox;
+    env.win.AMP_MODE.test = true;
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  function mockRtcExecution(rtcResponse, element, opt_textFunction) {
+    impl = new AmpAdNetworkDoubleclickImpl(element, env.win.document, env.win);
+    let textFunction = () => {
+      return Promise.resolve(JSON.stringify(rtcResponse));
+    };
+    textFunction = opt_textFunction || textFunction;
+    xhrMock.returns(
+        Promise.resolve({
+          redirected: false,
+          status: 200,
+          text: textFunction,
+        })
+      );
+    impl.populateAdUrlState();
+    return impl.executeRtc_(env.win.document);
+  }
+
+  function setRtcConfig(rtcConfigJson) {
+    const rtcConf = env.win.document.getElementById('amp-rtc');
+    rtcConf.innerText = JSON.stringify(rtcConfigJson);
+  }
+
+  beforeEach(() => {
+    resetRtcStateForTesting();
+    return createIframePromise().then(fixture => {
```

Comment:
Ditto... realWin shouldn't require wrapping in an iframe anymore and should load all of the services.  Instead you should refer to env.win & env.win.document